File: AdvantageFramework/Database/Procedures/AlertCategory.vb
Namespace Database.Procedures.AlertCategory

    <HideModuleName()> _
    Public Module Methods

#Region " Constants "



#End Region

#Region " Enum "



#End Region

#Region " Variables "



#End Region

#Region " Properties "



#End Region

#Region " Methods "

        Public Function LoadByAlertTypeIDAndAlertCategoryDescription(ByVal DbContext As Database.DbContext, ByVal AlertTypeID As String, ByVal AlertCategoryDescription As String) As Database.Entities.AlertCategory

            Try

                LoadByAlertTypeIDAndAlertCategoryDescription = (From AlertCategory In DbContext.AlertCategories
                                                                Where AlertCategory.Description = AlertCategoryDescription AndAlso
                                                                      AlertCategory.AlertTypeID = AlertTypeID
                                                                Select AlertCategory).SingleOrDefault

            Catch ex As Exception
                LoadByAlertTypeIDAndAlertCategoryDescription = Nothing
            End Try

        End Function
        Public Function LoadByAlertTypeID(ByVal DbContext As Database.DbContext, ByVal AlertTypeID As String) As System.Data.Entity.Infrastructure.DbQuery(Of Database.Entities.AlertCategory)

            LoadByAlertTypeID = From AlertCategory In DbContext.AlertCategories
                                Where AlertCategory.AlertTypeID = AlertTypeID
                                Select AlertCategory

        End Function
        Public Function LoadUserDefined(ByVal DbContext As AdvantageFramework.Database.DbContext) As IQueryable(Of AdvantageFramework.Database.Entities.AlertCategory)

            LoadUserDefined = (From AlertCategory In DbContext.AlertCategories
                               Where AlertCategory.IsUserDefined = True
                               Order By AlertCategory.Description
                               Select AlertCategory)


        End Function
        Public Function LoadUserDefinedActive(ByVal DbContext As AdvantageFramework.Database.DbContext) As IQueryable(Of AdvantageFramework.Database.Entities.AlertCategory)

            LoadUserDefinedActive = (From AlertCategory In DbContext.AlertCategories
                                     Where AlertCategory.IsUserDefined = True AndAlso (AlertCategory.IsInactive Is Nothing OrElse AlertCategory.IsInactive = 0)
                                     Order By AlertCategory.Description
                                     Select AlertCategory)


        End Function
        Public Function LoadByID(ByVal DbContext As Database.DbContext, ByVal AlertCategoryID As Integer) As Database.Entities.AlertCategory

            Try

                LoadByID = (From AlertCategory In DbContext.AlertCategories
                            Where AlertCategory.ID = AlertCategoryID
                            Select AlertCategory).SingleOrDefault

            Catch ex As Exception
                LoadByID = Nothing
            End Try

        End Function
        Public Function Load(ByVal DbContext As Database.DbContext) As System.Data.Entity.Infrastructure.DbQuery(Of Database.Entities.AlertCategory)

            Load = From AlertCategory In DbContext.AlertCategories
                   Select AlertCategory

        End Function
        Public Function Insert(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal AlertCategory As AdvantageFramework.Database.Entities.AlertCategory, ByRef ErrorMessage As String) As Boolean

            'objects
            Dim Inserted As Boolean = False
            Dim IsValid As Boolean = True
            Dim NextID As Integer = 0

            Try

                Try

                    NextID = CType(DbContext.Database.SqlQuery(Of Integer)("SELECT ISNULL(MAX(ALERT_CAT_ID), 0) + 1 FROM ALERT_CATEGORY WITH(NOLOCK);").FirstOrDefault, Integer)

                Catch ex As Exception
                    NextID = 0
                End Try

                If NextID > 0 Then

                    AlertCategory.ID = NextID

                    DbContext.AlertCategories.Add(AlertCategory)

                    ErrorMessage = AlertCategory.ValidateEntity(IsValid)

                    If IsValid Then

                        DbContext.SaveChanges()

                        Inserted = True

                    End If

                End If
            Catch ex As Exception

                Inserted = False

                ErrorMessage = ex.Message

                If ex.InnerException IsNot Nothing Then

                    ErrorMessage &= ex.InnerException.Message.ToString()

                End If

            Finally
                Insert = Inserted
            End Try

        End Function
        Public Function Update(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal AlertCategory As AdvantageFramework.Database.Entities.AlertCategory, ByRef ErrorMessage As String) As Boolean

            'objects
            Dim Updated As Boolean = False
            Dim IsValid As Boolean = True

            Try

                DbContext.Entry(AlertCategory).State = Entity.EntityState.Modified

                ErrorMessage = AlertCategory.ValidateEntity(IsValid)

                If IsValid Then

                    DbContext.SaveChanges()

                    Updated = True

                End If

            Catch ex As Exception

                Updated = False

                ErrorMessage = ex.Message

                If ex.InnerException IsNot Nothing Then

                    ErrorMessage &= System.Environment.NewLine & System.Environment.NewLine & ex.InnerException.Message

                End If

            Finally
                Update = Updated
            End Try

        End Function
        Public Function Delete(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal AlertCategory As AdvantageFramework.Database.Entities.AlertCategory) As Boolean

            Dim Deleted As Boolean = False
            Dim IsValid As Boolean = True
            Dim ErrorText As String = ""

            Try

                If IsValid Then

                    DbContext.Entry(AlertCategory).State = Entity.EntityState.Deleted

                    DbContext.SaveChanges()

                    Deleted = True

                Else

                    AdvantageFramework.Navigation.ShowMessageBox(ErrorText)

                End If

            Catch ex As Exception
                Deleted = False
            Finally
                Delete = Deleted
            End Try

        End Function

#End Region

    End Module

End Namespace

File: AdvantageFramework.Core/Database/Procedures/AlertCategory.cs
using System;
using System.Linq;

namespace AdvantageFramework.Core.Database.Procedures
{
    public static partial class AlertCategory
    {
        #region  Constants 



        #endregion

        #region  Enum 



        #endregion

        #region  Variables 



        #endregion

        #region  Properties 



        #endregion

        #region  Methods 

        public static Database.Entities.AlertCategory LoadByAlertTypeIDAndAlertCategoryDescription(Database.DbContext DbContext, string AlertTypeID, string AlertCategoryDescription)
        {
            Database.Entities.AlertCategory LoadByAlertTypeIDAndAlertCategoryDescriptionRet = default;
            try
            {
                LoadByAlertTypeIDAndAlertCategoryDescriptionRet = (from AlertCategory in DbContext.AlertCategories.AsQueryable()
                                                                   where AlertCategory.AlertDesc == AlertCategoryDescription && AlertCategory.AlertTypeId.ToString() == AlertTypeID
                                                                   select AlertCategory).SingleOrDefault();
            }
            catch (Exception)
            {
                LoadByAlertTypeIDAndAlertCategoryDescriptionRet = default;
            }

            return LoadByAlertTypeIDAndAlertCategoryDescriptionRet;
        }

        public static System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> LoadByAlertTypeID(Database.DbContext DbContext, string AlertTypeID)
        {
            System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> LoadByAlertTypeIDRet = default;
            LoadByAlertTypeIDRet = from AlertCategory in DbContext.AlertCategories.AsQueryable()
                                   where AlertCategory.AlertTypeId.ToString() == AlertTypeID
                                   select AlertCategory;
            return LoadByAlertTypeIDRet;
        }

        public static System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> LoadUserDefined(AdvantageFramework.Core.Database.DbContext DbContext)
        {
            System.Linq.IQueryable < AdvantageFramework.Core.Database.Entities.AlertCategory> LoadUserDefinedRet = default;
            LoadUserDefinedRet = (from AlertCategory in DbContext.AlertCategories.AsQueryable()
                                  where AlertCategory.IsUserDefined == true
                                  orderby AlertCategory.AlertDesc
                                  select AlertCategory);
            return LoadUserDefinedRet;
        }

        public static System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> LoadUserDefinedActive(AdvantageFramework.Core.Database.DbContext DbContext)
        {
            System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> LoadUserDefinedActiveRet = default;
            LoadUserDefinedActiveRet = (from AlertCategory in DbContext.AlertCategories.AsQueryable()
                                        where AlertCategory.IsUserDefined == true && (AlertCategory.IsInactive == null || AlertCategory.IsInactive == false)
                                        orderby AlertCategory.AlertDesc
                                        select AlertCategory);
            return LoadUserDefinedActiveRet;
        }

        public static Database.Entities.AlertCategory LoadByID(Database.DbContext DbContext, int AlertCategoryID)
        {
            Database.Entities.AlertCategory LoadByIDRet = default;
            try
            {
                LoadByIDRet = (from AlertCategory in DbContext.AlertCategories.AsQueryable()
                               where AlertCategory.AlertCatId == AlertCategoryID
                               select AlertCategory).SingleOrDefault();
            }
            catch (Exception)
            {
                LoadByIDRet = default;
            }

            return LoadByIDRet;
        }

        public static System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> Load(Database.DbContext DbContext)
        {
            System.Linq.IQueryable<AdvantageFramework.Core.Database.Entities.AlertCategory> LoadRet = default;
            LoadRet = from AlertCategory in DbContext.AlertCategories.AsQueryable()
                      select AlertCategory;
            return LoadRet;
        }

        //public static bool Insert(AdvantageFramework.Core.Database.DbContext DbContext, AdvantageFramework.Core.Database.Entities.AlertCategory AlertCategory, ref string ErrorMessage)
        //{
        //    bool InsertRet = default;

        //    // objects
        //    bool Inserted = false;
        //    bool IsValid = true;
        //    int NextID = 0;
        //    try
        //    {
        //        try
        //        {
        //            NextID = Conversions.ToInteger(DbContext.Database.SqlQuery<int>("SELECT ISNULL(MAX(ALERT_CAT_ID), 0) + 1 FROM ALERT_CATEGORY WITH(NOLOCK);").FirstOrDefault);
        //        }
        //        catch (Exception)
        //        {
        //            NextID = 0;
        //        }

        //        if (NextID > 0)
        //        {
        //            AlertCategory.ID = NextID;
        //            DbContext.AlertCategories.Add(AlertCategory);
        //            ErrorMessage = AlertCategory.ValidateEntity(IsValid);
        //            if (IsValid)
        //            {
        //                DbContext.SaveChanges();
        //                Inserted = true;
        //            }
        //        }
        //    }
        //    catch (Exception ex)
        //    {
        //        Inserted = false;
        //        ErrorMessage = ex.Message;
        //        if (ex.InnerException is object)
        //        {
        //            ErrorMessage += ex.InnerException.Message.ToString();
        //        }
        //    }
        //    finally
        //    {
        //        InsertRet = Inserted;
        //    }

        //    return InsertRet;
        //}

        //public static bool Update(AdvantageFramework.Core.Database.DbContext DbContext, AdvantageFramework.Core.Database.Entities.AlertCategory AlertCategory, ref string ErrorMessage)
        //{
        //    bool UpdateRet = default;

        //    // objects
        //    bool Updated = false;
        //    bool IsValid = true;
        //    try
        //    {
        //        DbContext.Entry(AlertCategory).State = Entity.EntityState.Modified;
        //        ErrorMessage = AlertCategory.ValidateEntity(IsValid);
        //        if (IsValid)
        //        {
        //            DbContext.SaveChanges();
        //            Updated = true;
        //        }
        //    }
        //    catch (Exception ex)
        //    {
        //        Updated = false;
        //        ErrorMessage = ex.Message;
        //        if (ex.InnerException is object)
        //        {
        //            ErrorMessage += Environment.NewLine + Environment.NewLine + ex.InnerException.Message;
        //        }
        //    }
        //    finally
        //    {
        //        UpdateRet = Updated;
        //    }

        //    return UpdateRet;
        //}

        //public static bool Delete(AdvantageFramework.Core.Database.DbContext DbContext, AdvantageFramework.Core.Database.Entities.AlertCategory AlertCategory)
        //{
        //    bool DeleteRet = default;
        //    bool Deleted = false;
        //    bool IsValid = true;
        //    string ErrorText = "";
        //    try
        //    {
        //        if (IsValid)
        //        {
        //            DbContext.Entry(AlertCategory).State = Entity.EntityState.Deleted;
        //            DbContext.SaveChanges();
        //            Deleted = true;
        //        }
        //        else
        //        {
        //            AdvantageFramework.Navigation.ShowMessageBox(ErrorText);
        //        }
        //    }
        //    catch (Exception ex)
        //    {
        //        Deleted = false;
        //    }
        //    finally
        //    {
        //        DeleteRet = Deleted;
        //    }

        //    return DeleteRet;
        //}

        #endregion
    }
}
